javascript/canvas.js

```diff
@@ -1,27 +1,93 @@
 class HangmanCanvas {
   constructor(secretWord) {
     this.context = document.getElementById('hangman').getContext('2d');
-    // ... your code goes here
+    this.secretWord = secretWord;// ... your code goes here
   }
 
   createBoard() {
-    // ... your code goes here
+    this.context.clearRect(0,0,this.context.width, this.context.height);
+    this.drawLines()
   }
 
   drawLines() {
-    // ... your code goes here
+    for(let i =0; i< this.secretWord.length; i++){
+      this.context.beginPath();
+      this.context.moveTo(500 + (i * 50), 400);
+      this.context.lineTo(530 + (i * 50), 400);
+      this.context.stroke();
+    }
   }
 
   writeCorrectLetter(index) {
-    // ... your code goes here
+    const guessedLetter = this.secretWord[index];
+
+    for(let i = 0; i < this.secretWord.length; i++) {
+      if(this.secretWord[i] === guessedLetter) {
+        this.context.font = "30px Arial"
+        this.context.fillText(guessedLetter.toUpperCase(), 505 + (i * 50), 380);
+      }
+    }   
   }
 
   writeWrongLetter(letter, errorsLeft) {
-    // ... your code goes here
+    const wrLet = 10 - errorsLeft;
+    
+    if(!this.secretWord.includes(letter)){ 
+      this.context.font = "30px Arial"
+      this.context.fillText(letter.toUpperCase(), 500 + (wrLet * 50), 100)
+    }
   }
 
   drawHangman(errorsLeft) {
-    // ... your code goes here
```

Comment:
the way this method is written, it will execute all the `if` statements (the part that checks the conditions) every time it is called.

we can avoid that, and reduce the extra `if` executions (checking of conditions) in a few different ways.

## 1: Return early

```js
drawHangman(errorsLeft) {
  if (errorsLeft === 9) {
    // ... do your thing
    return; // same as `return undefined`
  }

  if (errorsLeft === 8) {
    // ... do your thing
    return;
  }
}
```

As soon as we `return`, we stop executing any further code in that method for that particular call.

This way, the moment an `if` statement in this method matches the condition and executes, the other `if` statements below it will not be executed any more. Since by then, we have done what we wanted to accomplish from this method already.

## 2: If/else

The same result can be achieved with if/else:

```js
drawHangman(errorsLeft) {
  if (errorsLeft === 9) {
    // ... do your thing
  } else if (errorsLeft === 8) {
    // ... do your thing
  } else if (...) {
    // ...
  }
}
```

## 3: Switch statements

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Very similar to if/else, but suitable when we are checking against one variable only. Which fits out case very well, since we are only interested in the value of `errorsLeft`.

```js
drawHangman(errorsLeft) {
  switch (errorsLeft) {
    case 9:
      // ... do your thing
      break;
    case 8:
      // ... do your think
      break;
    default:
      // errorsLeft didn't match any cases
  }
}
```

---

All 4 options (3 in the comment, and the code that's already available in the PR) will end up producing the same result anyways.

These concerns become more important in projects where being highly performant becomes a key requirement.

Reply:
I couldn't believe my eyes when I saw the notification :D what are you doing here?!? call me back :)

ps. and of course, thanks for the feedback, I'll update